docs/ebpf/guides/getting-started.md

```diff
@@ -0,0 +1,311 @@
+# Creating an eBPF-powered Go application
+
+!!! tip ""
+    For a high-level understanding of what eBPF is and how it works, please see
+    [the eBPF introduction at ebpf.io](https://ebpf.io/what-is-ebpf). This guide
+    assumes familiarity with the basic concepts and terminology, as well as a
+    basic understanding of the Go toolchain (`go build`). We'll also rely on `go
+    generate`, see the [Go blog on Generating
+    Code](https://go.dev/blog/generate) if you're not yet familiar.
+
+    Discover [more projects using {{ proj }} here](../users.md). The main
+    repository also contains an [examples
+    directory](https://github.com/cilium/ebpf/tree/main/examples) with minimal
+    demo applications that can be tested on any supported Linux machine.
+
+In this guide, we'll walk you through building a new eBPF-powered Go application
+from scratch. We'll introduce the toolchain, write a minimal eBPF C example and
+compile it using [bpf2go](../bpf2go/index.md). Then, we'll put together a Go
+application that loads the eBPF program into the kernel and periodically
+displays its output.
+
+The application attaches an eBPF program to an XDP hook that counts the number
+of packets received by a physical interface. Filtering and modifying packets is
+a major use case for eBPF, so you'll see a lot of its features being geared
+towards it. However, eBPF's capabilities are ever-growing, and it has been
+adopted for tracing, systems and application observability, security and much
+more.
+
+## eBPF C program
+
+!!! abstract "Dependencies"
+    To follow along with the example, you'll need:
+
+    * Linux kernel version 5.7 or later, for bpf_link support
+    * LLVM 7 or later
+    * Go compiler version supported by {{ proj }}'s Go module
+    * libbpf headers, typically in distro packages `libbpf` or `libbpf-dev`
+      (Debian/Ubuntu)
+
+Let's begin by writing our eBPF C program, as its structure will be used as the
+basis for generating Go boilerplate.
+
+Click the :material-plus-circle: annotations in the code snippet for a detailed
+explanation of the individual components.
+
+{{ c_example('getting_started_counter', title='counter.c') }}
+
+1. When putting C files alongside Go files, they need to be excluded by a Go
+   build tag, otherwise `go build` will complain with `C source files not
+   allowed when not using cgo or SWIG`. The Go toolchain can safely ignore our
+   eBPF C files.
+
+1. Include headers containing the C macros used in the example. Identifiers such
```

Comment:
I thought about this more, and I'd prefer if these are numbered correctly. Using `1.` everywhere makes it harder to correlate the hint in the source code with the explanation here.

Your argument that using `1.` is better because it makes reordering easier doesn't make sense to me: you also need to renumber the whole source if you do that. So the best thing is to always add to the end of the list anyways.

Reply:
Being able to use 1. for all entries in numbered lists is just a markdown thing. We don't have to keep it that way, just mighty convenient when iterating.